src/fsharp/symbols/Symbols.fs

```diff
@@ -2058,6 +2058,11 @@ and FSharpMemberOrFunctionOrValue(cenv, d:FSharpMemberOrValData, item) =
         match d with
         | V valRef -> not (SymbolHelpers.isFunction cenv.g valRef.Type)
         | _ -> false
+        
+    member x.IsValUnused =
+        match d with
+        | V v -> v.HasBeenReferenced && not v.IsCompiledAsTopLevel && not (v.DisplayName.StartsWithOrdinal("_")) && not v.IsCompilerGenerated
```

Comment:
Maybe just have a `HasBeenReferenced` property?
Also: `DisplayName` is much slower than `LogicalName` since it tried to do a lot of additional stuff. Anyway for us it's better to get the name from out tree.